src/overlays/actors/ovl_En_Gs/z_en_gs.c

```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EV_BUTTERFRY_TO_FAIRY);
+                        Actor_SetSwitchFlag(globalCtx, this->unk_196);
+                    }
+                    break;
+
+                case 10:
+                    if (!Flags_GetSwitch(globalCtx, this->unk_196)) {
+                        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF, this->actor.world.pos.x,
+                                    this->actor.world.pos.y + 40.0f, this->actor.world.pos.z, 0, 0, 0, 7);
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EV_BUTTERFRY_TO_FAIRY);
+                        Actor_SetSwitchFlag(globalCtx, this->unk_196);
+                    }
+                    break;
+
+                case 0:
+                    if ((this->actor.params == ENGS_1) && (gSaveContext.playerForm == PLAYER_FORM_DEKU)) {
+                        this->unk_194 = 1;
+                        this->unk_19C = 5;
+                        this->unk_19A |= 1;
+                        func_80999AC0(this);
+                        func_809984F4(this, globalCtx);
+                    }
+                    break;
+
+                case 2:
+                    if ((this->actor.params == ENGS_1) && (gSaveContext.playerForm == PLAYER_FORM_ZORA)) {
+                        this->unk_194 = 3;
+                        this->unk_19C = 5;
+                        this->unk_19A |= 1;
+                        func_80999AC0(this);
+                        func_809984F4(this, globalCtx);
+                    }
+                    break;
+
+                case 1:
+                    if ((this->actor.params == ENGS_1) && (gSaveContext.playerForm == PLAYER_FORM_GORON)) {
+                        this->unk_194 = 2;
+                        this->unk_19C = 5;
+                        this->unk_19A |= 1;
+                        func_80999AC0(this);
+                        func_809984F4(this, globalCtx);
+                    }
+                    break;
+            }
+            break;
+
+        case 0:
+        case 4:
+            func_80998300(this, globalCtx);
+
+        case 26:
+            func_80997D14(this, globalCtx);
+            break;
+    }
+}
+
+void func_80998300(EnGs* this, GlobalContext* globalCtx) {
+    if (this->actor.cutscene != -1) {
+        ActorCutscene_Stop(globalCtx->unk_1879C[0]);
+    }
+}
+
+f32 func_80998334(EnGs* this, GlobalContext* globalCtx, f32* arg2, f32* arg3, s16* arg4, f32 arg5, f32 arg6, f32 arg7,
+                  s32 arg8, s32 arg9) {
+    f32 sp2C = Math_SmoothStepToF(arg2, *arg3, arg5, arg6, arg7);
+
+    if (arg9 == 0) {
+        sp2C = Math_SmoothStepToF(arg2, *arg3, arg5, arg6, arg7);
+        this->unk_1B0[0].x = (__sinf((*arg4 % arg8) * (1.0f / arg8) * 360.0f * 0.017453292f) * *arg2) + 1.0f;
+        this->unk_1B0[0].y = 1.0f - (__sinf((*arg4 % arg8) * (1.0f / arg8) * 360.0f * 0.017453292f) * *arg2);
```

Comment:
```suggestion
        this->unk_1B0[0].x = (__sinf(DEGF_TO_RADF((*arg4 % arg8) * (1.0f / arg8) * 360.0f)) * *arg2) + 1.0f;
        this->unk_1B0[0].y = 1.0f - (__sinf(DEGF_TO_RADF((*arg4 % arg8) * (1.0f / arg8) * 360.0f)) * *arg2);
```
If `__sinf((*arg4 % arg8) * (1.0f / arg8) * 2 * M_PI)` works, then even better